### PR TITLE
Add Asterisks on Child Registration page on Mandatory

### PR DESCRIPTION
**Pull request by sherrif10**
Follow up pr  https://sd-vxnaid.atlassian.net/browse/VRVU-229  cc @jdolkowski @druchniewicz . Am still improving this.

### CODE REVIEW

File: app/src/main/res/values/strings.xml
Comment (druchniewicz) [inline app/src/main/res/values/strings.xml:1]: why different approach than in other fields?

**Review by druchniewicz — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/jnj/vaccinetracker/register/screens/RegisterParticipantParticipantDetailsFragment.kt
Comment (druchniewicz) [inline app/src/main/java/com/jnj/vaccinetracker/register/screens/RegisterParticipantParticipantDetailsFragment.kt:1]: @sherrif10 this PR is related to "Add Asterisk on Child registration page". Please finish this task and PR first. (I left you comment, why did you apply another approach than in other fields? please use the same approach). 

After that, please open a new PR related to something different - to birth weight validator. I see in Jira board that there is a separate task for it, so please create a separate PR for that.

**Review by druchniewicz — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/jnj/vaccinetracker/register/screens/RegisterParticipantParticipantDetailsFragment.kt
Comment (sherrif10) [reply app/src/main/java/com/jnj/vaccinetracker/register/screens/RegisterParticipantParticipantDetailsFragment.kt:1]: Hello @druchniewicz , Am pushing the latest changes in regards to your commits you had shared above.

**Comment by sherrif10** [comment]
Hey @druchniewicz ,  i addressed your comments + couple few fixes.

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -85,7 +86,7 @@
     <string name="iris_scan_right_title">Iris scan - Right eye</string>
     <string name="participant_matching_title">Child matching</string>
     <string name="participant_flow_child_number_title">Child Number</string>
-    <string name="participant_flow_participant_id_title">Child ID</string>
+    <string name="participant_flow_participant_id_title">Child ID*</string>
```
Comment (druchniewicz) [inline app/src/main/res/values/strings.xml]: @sherrif10 this property is used in whole app. How often do I have to repeat this?..

File: app/src/main/res/values/strings.xml
Comment (druchniewicz) [reply app/src/main/res/values/strings.xml:1]: @sherrif10 once again, why can't just use the same approach as you used in the other fields where you added aditional TextView for asterisk?

**Review by druchniewicz — CHANGES_REQUESTED** [review]

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -30,6 +30,7 @@
     <string name="settings_label_deviceId">Device ID</string>
     <string name="settings_label_no_licenses">No active licenses</string>
     <string name="settings_label_deactivate">Deactivate</string>
+    <string name="label_gender_with_asterisk" translatable="false">Gender</string>
```
Comment (druchniewicz) [inline app/src/main/res/values/strings.xml:33]: this is not needed, Gender string already exists

File: app/src/main/res/layout/fragment_register_participant_participant_details.xml [hunk 3]
```diff
@@ -560,22 +561,37 @@
...
-                    android:text="@string/participant_registration_details_label_gender"
-                    android:textAppearance="@style/LabelText" />
+                    android:orientation="horizontal">
+
+                    <TextView
+                        android:id="@+id/label_gender"
+                        android:layout_width="wrap_content"
+                        android:layout_height="wrap_content"
+                        android:maxLines="1"
+                        android:text="@string/label_gender_with_asterisk"
```
Comment (druchniewicz) [inline app/src/main/res/layout/fragment_register_participant_participant_details.xml:574]: please dont change this line and use participant_registration_details_label_gender as it was before

File: app/src/main/res/layout/fragment_register_participant_participant_details.xml [hunk 4]
```diff
@@ -93,7 +101,7 @@
                     android:layout_weight="3"
                     android:text="@string/generate_Qr_code"
                     android:textAppearance="@style/ButtonTextStyling"
-                    android:visibility = "visible"/>
+                    android:visibility ="@{viewModel.participantUuid != null ? View.GONE : View.VISIBLE}"/>
```
Comment (druchniewicz) [inline app/src/main/res/layout/fragment_register_participant_participant_details.xml:104]: what is the purpose of this line? Shouldn't this button always be visible?

File: app/src/main/java/com/jnj/vaccinetracker/register/screens/RegisterParticipantParticipantDetailsFragment.kt
Comment (druchniewicz) [inline app/src/main/java/com/jnj/vaccinetracker/register/screens/RegisterParticipantParticipantDetailsFragment.kt:1]: please remove it. you changed this code in another pull request

**Review by druchniewicz — CHANGES_REQUESTED** [review]

**Comment by sherrif10** [comment]
i will open a new PR

File: app/src/main/res/layout/fragment_register_participant_participant_details.xml (hunk 4 above)
Comment (sherrif10) [reply app/src/main/res/layout/fragment_register_participant_participant_details.xml:104]: Resolved.

**Pull request closed without merging**